rules/linux/persistence_pluggable_authentication_module_creation.toml

```diff
@@ -0,0 +1,94 @@
+[metadata]
+creation_date = "2024/03/06"
+integration = ["endpoint"]
+maturity = "production"
+updated_date = "2024/03/06"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule monitors for the creation or modification of Pluggable Authentication Module (PAM) shared object files or
+configuration files. Attackers may create or modify these files to maintain persistence on a compromised system, or 
+harvest account credentials.
+"""
+false_positives = [
+    "Trusted system module updates or allowed Pluggable Authentication Module (PAM) daemon configuration changes.",
+]
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Creation or Modification of Pluggable Authentication Module or Configuration"
+references = [
+    "https://github.com/zephrax/linux-pam-backdoor",
+    "https://github.com/eurialo/pambd",
+    "http://0x90909090.blogspot.com/2016/06/creating-backdoor-in-pam-in-5-line-of.html",
+    "https://www.trendmicro.com/en_us/research/19/i/skidmap-linux-malware-uses-rootkit-capabilities-to-hide-cryptocurrency-mining-payload.html",
+]
+risk_score = 47
+rule_id = "f48ecc44-7d02-437d-9562-b838d2c41987"
+severity = "medium"
+tags = [
+    "Domain: Endpoint",
+    "OS: Linux",
+    "Use Case: Threat Detection",
+    "Tactic: Credential Access",
+    "Tactic: Persistence",
+    "Data Source: Elastic Defend",
+]
+timestamp_override = "event.ingested"
+type = "eql"
+query = '''
+file where host.os.type == "linux" and event.action in ("rename", "creation") and (
+  (file.path : ("/lib/security/*", "/lib64/security/*", "/usr/lib/security/*", "/usr/lib64/security/*",
+  "/usr/lib/x86_64-linux-gnu/security/*") and file.extension == "so") or
+  (file.path : "/etc/pam.d/*" and file.extension == null) or 
+  (file.path in ("/etc/security/pam_env", "/etc/pam.conf"))
```

Comment:
Should we broaden `pam_env` to `pam_*`?

Reply:
I agree, I was not aware of the fact that some Linux distributions leverage other PAM configuration files there. E.g. after some digging, I noticed that (for example in https://superuser.com/questions/962683/in-linux-what-is-etc-security) it would be good to increase coverage. Ty!